compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10563,69 +10564,53 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
-            {
-            if (baseSource1Ref->getIndexRegister())
-                  numDeps++;
-            if (baseSource1Ref->getBaseRegister())
-                  numDeps++;
-            if (baseSource2Ref->getIndexRegister() && (baseSource2Ref->getIndexRegister() != baseSource1Ref->getIndexRegister()))
-                  numDeps++;
-            if (baseSource2Ref->getBaseRegister() && (baseSource2Ref->getBaseRegister() != baseSource1Ref->getBaseRegister()))
-                  numDeps++;
-            }
-
-         if (isFoldedIf)
-            {
-            if (deps == NULL)
-               regDeps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, numDeps, cg);
-            else
-               regDeps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(deps, 0, numDeps, cg);
-            }
-         else
-            {
-            if (needResultReg)
-               {
-               numDeps++; // for retValReg
-               regDeps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, numDeps, cg);
-               regDeps->addPostCondition(retValReg, TR::RealRegister::AssignAny);
-               }
-            }
-
-         if (length > (earlyCLCEnabled ? earlyCLCThreshold : TR_MAX_CLC_SIZE))
+         if (needResultReg)
             {
-            if (baseSource1Ref->getIndexRegister())
-               regDeps->addPostConditionIfNotAlreadyInserted(baseSource1Ref->getIndexRegister(), TR::RealRegister::AssignAny);
-
-            if (baseSource1Ref->getBaseRegister())
-               regDeps->addPostConditionIfNotAlreadyInserted(baseSource1Ref->getBaseRegister(), TR::RealRegister::AssignAny);
-
-            if (baseSource2Ref->getIndexRegister() && (baseSource2Ref->getIndexRegister() != baseSource1Ref->getIndexRegister()))
-               regDeps->addPostConditionIfNotAlreadyInserted(baseSource2Ref->getIndexRegister(), TR::RealRegister::AssignAny);
-
-            if (baseSource2Ref->getBaseRegister() && (baseSource2Ref->getBaseRegister() != baseSource1Ref->getBaseRegister()))
-               regDeps->addPostConditionIfNotAlreadyInserted(baseSource2Ref->getBaseRegister(), TR::RealRegister::AssignAny);
+            numDeps++; // for retValReg
+            regDeps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, numDeps, cg);
+            regDeps->addPostCondition(retValReg, TR::RealRegister::AssignAny);
+            generateRRInstruction(cg, TR::InstOpCode::getXORRegOpCode(), node, retValReg, retValReg);
             }
          }
 
-      TR::LabelSymbol *condLabel = isCondLabelNeeded ? generateLabelSymbol(cg) : NULL;
+      if (baseSource1Ref->getBaseRegister())
+         regDeps->addPostConditionIfNotAlreadyInserted(baseSource1Ref->getBaseRegister(), TR::RealRegister::AssignAny);
 
-      TR::LabelSymbol *endLabel = isEndLabelNeeded ? generateLabelSymbol(cg) : NULL;
+      if (baseSource2Ref->getBaseRegister() && (baseSource2Ref->getBaseRegister() != baseSource1Ref->getBaseRegister()))
```

Comment:
I think this is not needed. If baseSource2Ref's base register is not null, call to addPostConditionIfNotAlreadyInserted would make sure that if the basereg for first and second mem ref is same, it will not be added.